core/validators/topology_snapshot.go

```diff
@@ -144,6 +145,22 @@ func (t *Topology) serialisePendingEthereumKeyRotation() []*snapshot.PendingEthe
 	return pkrs
 }
 
+func (t *Topology) serialiseUnresolvedEthereumKeyRotations() []*snapshot.PendingEthereumKeyRotation {
+	ukrs := make([]*snapshot.PendingEthereumKeyRotation, 0, len(t.unresolvedEthKeyRotations))
+	for _, r := range t.unresolvedEthKeyRotations {
+		ukrs = append(ukrs, &snapshot.PendingEthereumKeyRotation{
+			NodeId:     r.NodeID,
+			NewAddress: r.NewAddress,
+			Submitter:  r.SubmitterAddress,
+			OldAddress: r.OldAddress,
+		})
+	}
+
+	sort.SliceStable(ukrs, func(i, j int) bool { return ukrs[i].GetNodeId() < ukrs[j].GetNodeId() })
```

Comment:
Can a node have multiple key rotation at a time again? If yes this is not gonna be enough I suppose and we'll need to sort over the nodeId an e.g the new address

Reply:
No they can't. This whole "unresolved rotation" check is to make sure that people can't stack up lots of ethereum key rotations and generate 100s of valid add signer bundles for themselves for 100s of different keys.